Bixal/move.mil
# Feature/158296845/add video field

Builds on top of #262 (including the commits in #262 that create a WYSIWYG-based option) with a new commit adding a dedicated field/paragraph for a video embed. I worked through the UX considerations with Phil and we decided to have the user add the public link to the video rather than an embed code that often includes style/layout opinions we may not want. The new preprocess and custom twig logic takes care of the dirty work converting the link into an embedded video.

On another note: I wrote a preprocess function (with Abby's help) two work days after giving a conference talk about how I never want to write preprocess functions. There's some glorious irony in that.

## Checklist

I have…

      against my changes

## Summary of Changes

This pull request…

- Creates a new `Video Embed` paragraph type and adds it to the list of supported paragraph types in our main content area.
- Creates a single "Video URL" field within that paragraph type
- Places help text on that field's entry explaining the user should add a public link rather than the embed code and that it supports Youtube and Vimeo.
- Adds a preprocess function that alchemizes the public link (if and only if entered properly) into a raw ID, with different processes for Youtube and Vimeo.
- Adds a Twig partial building the proper `<iframe>` embed code around that ID, again differently for each video service.

## Testing

To verify the changes proposed in this pull request…

1. Pull code and import config
1. Build theme assets (if not already done for #262)
1. Go to a page and add a new paragraph. Select the `Video Embed` option.
1. Paste the example Vimeo url listed in the help text into the field
1. Create a new paragraph and do the same with the example Youtube url.
1. Save the page. The videos should render properly.

## Screenshots

Local example of rendered videos:
<img width="619" alt="video embed field - local examples" src="https://user-images.githubusercontent.com/29130580/44752084-f6d32480-aae7-11e8-8266-a06e077c53d9.png">

Local example(s) of content editor UI (note both Youtube versions work):
<img width="759" alt="video embed field - content editor ui with examples" src="https://user-images.githubusercontent.com/29130580/44752175-46195500-aae8-11e8-8535-f22a36e0e956.png">
